api/cedar_metadata_records/permissions.py

```diff
@@ -0,0 +1,29 @@
+import logging
+
+from rest_framework import permissions
+
+from api.base.utils import get_user_auth
+
+from osf.models import BaseFileNode, CedarMetadataRecord, Node, Registration
+
+logger = logging.getLogger(__name__)
+
+
+class CedarMetadataRecordPermission(permissions.BasePermission):
+
+    def has_object_permission(self, request, view, obj):
+
+        assert isinstance(obj, CedarMetadataRecord), 'obj must be a CedarMetadataRecord'
+
+        auth = get_user_auth(request)
+
+        delegated_object = obj.guid.referent
+        if isinstance(delegated_object, BaseFileNode):
+            delegated_object = delegated_object.target
+        elif not isinstance(delegated_object, Node) and not isinstance(delegated_object, Registration):
```

Comment:
nit: you can merge this into not isinstance(delegated_object, (Node, Registration))

Reply:
Ah, thanks for the tip, exactly what I need 👍